packages/playground-examples/copy/ko/JavaScript/Modern JavaScript/Import Export.ts

```diff
@@ -0,0 +1,109 @@
+//// { order: 1, target: "ES5" }
+
+// JavaScript는 2016년에 import/export를 다시 추가하였고
+// TypeScript는 파일 간의 연결과 외부 모듈로의 연결에 관한
+// 이 스타일의 지원을 마쳤습니다.
+// TypeScript는 코드와 함꼐 전달될 타입들을 허용함으로써
+// 이 문법을 확장시킵니다.
+
+// 모듈에서 코드를 import해오는 것을 봅시다.
+
+import { danger, message, warn, DangerDSLType } from "danger";
+
+// danger라는 node 모듈에서 일련의 이름 붙은 import를 가져옵니다.
+// 4개 이상의 import를 할 수 있지만,
+// 이것들이 우리가 import하기로 선택한 것들입니다.
+
+// 여러분이 무엇을 import하는지 이름을 구체적으로 적으면
+// 앱에서 사용되지 않는 코드를 제거할 수 있으며
+// 특정 파일에서 무엇이 사용되는지
+// 쉽게 이해할 수 있습니다.
+
+// 이 경우: danger, message, warn은 JavaScript import입니다.
+// 반면 DangerDSLType은 인터페이스 타입입니다.
+
+// TypeScript는 JSDoc을 이용하여 개발자의 코드를 문서화시켜주며,
+// 문서들은 또 import됩니다.
+// 예를 들어 만약 아래의 서로 다른 부분들에 커서를 올리면,
+// 각각에 대한 설명을 볼 수 있습니다.
+
+danger.git.modified_files;
+
+// 이러한 문서 표기를 제공하는 방법을 알고 싶다면
+// example:jsdoc-support를 참고하세요.
+
+// 코드를 import하는 다른 방식은
+// 모듈의 default export를 사용하는 것입니다.
+// 이 예제는 모듈 디버깅인데, 로그 기능을 하는 함수를 내보냅니다.
+
+import debug from "debug";
+const log = debug("playground");
+log("Started running code");
+
+// default export는 실제 이름이 없기 때문에,
+// TypeScript의 리팩토링 지원과 같은
+// 정적 분석 도구가 적용되었을 때 까다로울 수 있지만
+// 그 사용법이 있습니다.
+
+// JavaScript의 importing/exporting 코드에는 오랜 역사가 있기 때문에,
+// default exports에는 헷갈리는 부분이 있습니다.
+// 몇몇 export에 관한 문서들은
+// import를 아래와 같은 식으로 쓸 수 있음을 보여줍니다.
+
+import req from "request";
+
+// 하지만 이는 실패하는데, stack overflow를 찾아보면
+// import를 다음과 같이 제안합니다.
+
+import * as req from "request";
+
+// 그리고 이는 작동합니다. 왜일까요?
+// exporting에 대한 섹션의 끝부분에서 다시 확인해보겠습니다.
+
+// import하기 위해서는, export가 가능해야 합니다.
+// export를 쓰는 최신 방식은 export 키워드를 쓰는 것입니다.
+
+/** 현재 롤에 남아 있는 스티커 수 */
+export const numberOfStickers = 11;
+
+// 다른 파일에서 아래와 같이 import할 수 있습니다.
+//
+// import { numberOfStickers } from "./path/to/file"
+
+// 하나의 파일에 원하는 만큼 많은 import를 할 수 있습니다.
+// default export도 비슷합니다.
+
+/** 스티커를 만들어 줍니다. */
+const stickerGenerator = () => {};
+export default stickerGenerator;
+
+// 다른 파일에서 아래와 같이 import할 수 있습니다.
+//
+// import getStickers from "./path/to/file"
+//
+// 이름은 모듈을 사용하는 곳에서 붙입니다.
+
+// 이는 import의 유일한 타입은 아니고, 일반적인 최신의 코드일 뿐입니다.
+// 모듈 간 공유할 수 있는 코드의 모든 방식을 다루는 것은
```

Comment:
```suggestion
// 모듈 간 공유할 수 있는 코드의 모든 방식을 다루는 것은
```
공유는 의미가 부정확한 것 같은데 좋은 표현이 생각나질 않네요 😢

Reply:
(원문) `Covering all of the ways code can cross module boundaries is a very long topic in the handbook:`
-> `모듈 간의 경계를 뛰어넘을 수 있는 코드` 어떨까용

Reply:
좋은 것 같아요 :)